lib/charms/operator_libs_linux/v0/apt.py

```diff
@@ -1315,47 +1409,293 @@ def _parse(line: str, filename: str) -> DebianRepository:
             raise InvalidSourceError("An invalid sources line was found in %s!", filename)
 
     def add(self, repo: DebianRepository, default_filename: Optional[bool] = False) -> None:
-        """Add a new repository to the system.
-
-        Args:
-          repo: a `DebianRepository` object
-          default_filename: an (Optional) filename if the default is not desirable
+        """Add a new repository to the system. May have destructive or surprising behaviour.
+
+        Will write in the deb822 format if:
+            1. repo.filename has the '.sources' extension
+            2. repo.filename is unset (""), but /etc/apt/sources.list.d/ubuntu.sources exists
+        Otherwise the one-per-line style will be used.
+
+        WARNING: the default_filename keyword argument is provided for backwards compatibility
+        only. It is not used, and was not used in the previous revision of this library.
+
+        WARNING: in the one-per-line format case, will mutate repo.options to add the 'signed-by'
+        key if both options and gpg_file are truthy (for example if a gpg_key_filename and
+        non-empty options were provided at DebianRepository initialisation time). If options were
+        not provided or are empty, but a gpg_key is available, will silently fail to include it.
```

Comment:
This is a bug. Potentially easy to fix. Doesn't affect the one user of this method, as the repository they pass doesn't use either `gpg_key` or `options`.